url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParserTokenManager.java

```diff
@@ -0,0 +1,1654 @@
+   catch(java.io.IOException e) {
+      jjStopStringLiteralDfa_0(3, active0, active1);
+      return 4;
+   }
+   switch(curChar)
+   {
+      case 66:
+      case 98:
+         return jjMoveStringLiteralDfa5_0(active0, 0x200000000000000L, active1, 0L);
+      case 67:
+      case 99:
+         return jjMoveStringLiteralDfa5_0(active0, 0x800000000000000L, active1, 0L);
+      case 69:
+      case 101:
+         if ((active0 & 0x2000000000000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 61, 48);
+         else if ((active1 & 0x2L) != 0L)
+            return jjStartNfaWithStates_0(4, 65, 48);
+         else if ((active1 & 0x100L) != 0L)
+            return jjStartNfaWithStates_0(4, 72, 48);
+         else if ((active1 & 0x800L) != 0L)
+            return jjStartNfaWithStates_0(4, 75, 48);
+         return jjMoveStringLiteralDfa5_0(active0, 0x1000000052800L, active1, 0x20L);
+      case 71:
+      case 103:
+         return jjMoveStringLiteralDfa5_0(active0, 0x400000000L, active1, 0L);
+      case 72:
+      case 104:
+         return jjMoveStringLiteralDfa5_0(active0, 0L, active1, 0xc0L);
+      case 73:
+      case 105:
+         return jjMoveStringLiteralDfa5_0(active0, 0x400000820000L, active1, 0L);
+      case 76:
+      case 108:
+         return jjMoveStringLiteralDfa5_0(active0, 0x4000000000000000L, active1, 0L);
+      case 78:
+      case 110:
+         if ((active1 & 0x8L) != 0L)
+            return jjStartNfaWithStates_0(4, 67, 48);
+         return jjMoveStringLiteralDfa5_0(active0, 0x80000000L, active1, 0L);
+      case 80:
+      case 112:
+         if ((active0 & 0x40000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 30, 48);
+         break;
+      case 82:
+      case 114:
+         if ((active0 & 0x10000000000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 52, 48);
+         else if ((active0 & 0x20000000000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 53, 48);
+         return jjMoveStringLiteralDfa5_0(active0, 0x100a00001000L, active1, 0L);
+      case 83:
+      case 115:
+         if ((active0 & 0x40000000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 42, 48);
+         return jjMoveStringLiteralDfa5_0(active0, 0L, active1, 0x4L);
+      case 84:
+      case 116:
+         if ((active0 & 0x80000L) != 0L)
+            return jjStartNfaWithStates_0(4, 19, 48);
+         else if ((active0 & 0x8000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 27, 48);
+         else if ((active0 & 0x8000000000000000L) != 0L)
+            return jjStartNfaWithStates_0(4, 63, 48);
+         return jjMoveStringLiteralDfa5_0(active0, 0x2200000L, active1, 0x10L);
+      case 85:
+      case 117:
+         return jjMoveStringLiteralDfa5_0(active0, 0x1000000L, active1, 0L);
+      default :
+         break;
+   }
+   return jjStartNfa_0(3, active0, active1);
+}
+static private int jjMoveStringLiteralDfa5_0(long old0, long active0, long old1, long active1)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 61 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.